### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by IvanGuculyak**

### CODE REVIEW

File: app/customer.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    def price_trip(self, shop: Shop) -> float:
+        fuel_consumption = self.car["fuel_consumption"]
+        distance = self.distance(shop)
+        price_road = 2 * (fuel_consumption / 100) * distance * fuel_price
+
+        price_product = 0
+        for product_name in self.product_cart.keys():
+            count_product = self.product_cart[product_name]
+            price = shop.products[product_name]
+            price_product += count_product * price
```
Comment (ArsenPidhoretskyi) [inline app/customer.py:30]: Use sum function

File: app/customer.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+for customer in customers:
+    customers_list.append(
+        Customer(
+            name=customer["name"],
+            product_cart=customer["product_cart"],
+            location=customer["location"],
+            money=customer["money"],
+            car=customer["car"]
+        )
+    )
```
Comment (ArsenPidhoretskyi) [inline app/customer.py:49]: Do it as a list comprehension
Also, move it to main.py not good to do it not in function
The same with shops

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,51 @@
...
+        print(f"Thanks, {customer.name}, for your purchase!")
+
+        count_milk = customer.product_cart["milk"]
+        count_bread = customer.product_cart["bread"]
+        count_butter = customer.product_cart["butter"]
+        milk_value = count_milk * shop_cheapest_check.products["milk"]
+        bread_value = count_bread * shop_cheapest_check.products["bread"]
+        butter_value = count_butter * shop_cheapest_check.products["butter"]
+
+        total_value = milk_value + bread_value + butter_value
```
Comment (ArsenPidhoretskyi) [inline app/main.py:37]: Hardcoded data. Products can be any!
For example, I can buy also cake, ...

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/customer.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        pow_y_coordinate = pow(self.location[1] - shop.location[1], 2)
+        return math.sqrt(pow_x_coordinate + pow_y_coordinate)
+
+    def product_value(self, shop: Shop) -> float:
+        return sum(
+            [
+                self.product_cart[product_name] * shop.products[product_name]
+                for product_name in self.product_cart.keys()
+            ]
+        )
```
Comment (Vasyl-Smutok) [inline app/customer.py:27]: no need to create list and use `keys`
```suggestion
        return sum(
            self.product_cart[product_name] * shop.products[product_name]
            for product_name in self.product_car
        )

```

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,50 @@
...
+            continue
+
+        print(f"{customer.name} rides to {shop_cheapest_check.name}")
+        print()
+        print("Date: 04/01/2021 12:33:41")
+        print(f"Thanks, {customer.name}, for your purchase!")
+
+        print("You have bought: ")
+
+        for product_name in customer.product_cart.keys():
```
Comment (Vasyl-Smutok) [inline app/main.py:32]: Default `keys`
```suggestion
        for product_name in customer.product_cart:
```

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: app/customer.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        pow_y_coordinate = pow(self.location[1] - shop.location[1], 2)
+        return math.sqrt(pow_x_coordinate + pow_y_coordinate)
+
+    def product_value(self, shop: Shop) -> float:
+        return sum(
+            [
+                self.product_cart[product_name] * shop.products[product_name]
+                for product_name in self.product_cart
+            ]
+        )
```
Comment (Vasyl-Smutok) [inline app/customer.py:27]: no need to create list
```suggestion
        return sum(
            self.product_cart[product_name] * shop.products[product_name]
            for product_name in self.product_cart
        )

```

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Review by vsmutok — APPROVED** [review]
Good Job 👍

**Pull request closed without merging**